orchestra/utils/task_lifecycle.py

```diff
@@ -140,9 +140,17 @@ def get_role_from_counter(role_counter):
 
 
 def role_counter_required_for_new_task(task):
+    """
+    Given a task find a level of expert that is required
+    to be assigned for task.
+    """
     if task.status not in [Task.Status.AWAITING_PROCESSING,
                            Task.Status.PENDING_REVIEW]:
         raise TaskAssignmentError('Status incompatible with new assignment')
+    elif (task.status == Task.Status.AWAITING_PROCESSING and
+          (task.assignments.filter(assignment_counter=0)
+           .exclude(status=TaskAssignment.Status.FAILED).exists())):
+        raise TaskAssignmentError('Task is in incorrect state')
     assignments_count = task.assignments.count()
```

Comment:
I think you should update this to the query i suggested in the previous comments so we don't have to hit the DB so many times.